cmd/agent/main.go

```diff
@@ -1,3 +1,52 @@
 package main
 
-func main() {}
+import (
+	"fmt"
+	"github.com/forester2k/metrics/internal/service"
+	"net/http"
+	"time"
+)
+
+const defaultReportInterval = 10
+const defaultPollInterval = 2
+const defaultHost = "localhost:8080"
+
+func poll() {
+	//fmt.Println("polling", time.Now())
+	// TODO: logic
+}
+
+func report(metric service.Saver) {
+	endpoint := "http://" + flagRunAddr + "/update" + metric.Path()
+	client := &http.Client{}
+	request, err := http.NewRequest(http.MethodPost, endpoint, nil)
+	if err != nil {
+		panic(err)
```

Comment:
```suggestion
		return err
```
https://t.me/bushigo/10

Агент должен работать, пока ему это явно не запретят. Никакие коллизии и ошибки не должны его ронять; его задача - собирать метрики и отправлять, что бы ни происходило.